File: backend/src/main/java/com/gxs/bank/repository/BeneficiaryRepository.java
package com.gxs.bank.repository;

import com.gxs.bank.model.Beneficiary;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface BeneficiaryRepository extends JpaRepository<Beneficiary, UUID> {
    List<Beneficiary> findByUserId(UUID userId);
}

File: backend-python/com/gxs/bank/repository/BeneficiaryRepository.py
from com.gxs.bank.model.Beneficiary import Beneficiary
from com.gxs.bank.repository._base import BaseRepository


class BeneficiaryRepository(BaseRepository):
    def __init__(self, db):
        super().__init__(db, Beneficiary)

    def findByUserId(self, userId: str):
        return self.db.query(Beneficiary).filter(Beneficiary.userId == userId).all()
